lib/cylc/review.py

```diff
@@ -0,0 +1,917 @@
+            "script": cherrypy.request.script_name,
+            "method": "suites",
+            "no_fuzzy_time": no_fuzzy_time,
+            "user": user,
+            "is_option_on": (
+                names and shlex.split(str(names)) != ["*"] or
+                order is not None and order != "time_desc" or
+                per_page is not None and per_page != per_page_default
+            ),
+            "names": names,
+            "page": page,
+            "order": order,
+            "per_page": per_page,
+            "per_page_default": per_page_default,
+            "entries": [],
+        }
+        name_globs = ["*"]
+        if names:
+            name_globs = shlex.split(str(names))
+        # Get entries
+        sub_names = [
+            ".service", "log", "share", "work", "suite.rc"]
+        for dirpath, dnames, fnames in os.walk(
+                user_suite_dir_root, followlinks=True):
+            if dirpath != user_suite_dir_root and (
+                    any(name in dnames or name in fnames
+                        for name in sub_names)):
+                dnames[:] = []
+            else:
+                continue
+            item = os.path.relpath(dirpath, user_suite_dir_root)
+            if not any(fnmatch(item, glob_) for glob_ in name_globs):
+                continue
+            try:
+                data["entries"].append({
+                    "name": item,
+                    "info": {},
+                    "last_activity_time": (
+                        self.get_last_activity_time(user, item))})
+            except OSError:
+                pass
+
+        if order == "name_asc":
+            data["entries"].sort(key=lambda entry: entry["name"])
+        elif order == "name_desc":
+            data["entries"].sort(key=lambda entry: entry["name"], reverse=True)
+        elif order == "time_asc":
+            data["entries"].sort(self._sort_summary_entries, reverse=True)
+        else:  # order == "time_desc"
+            data["entries"].sort(self._sort_summary_entries)
+        data["of_n_entries"] = len(data["entries"])
+        if per_page:
+            data["n_pages"] = data["of_n_entries"] / per_page
+            if data["of_n_entries"] % per_page != 0:
+                data["n_pages"] += 1
+            offset = (page - 1) * per_page
+            data["entries"] = data["entries"][offset:offset + per_page]
+        else:
+            data["n_pages"] = 1
+
+        # Get basic ros(i)e suite info (project and title) per entry if found
+        for entry in data["entries"]:
+            user_suite_dir = os.path.join(user_suite_dir_root, entry["name"])
+            rosie_suite_info = os.path.join(user_suite_dir, "rose-suite.info")
+            if os.path.isfile(rosie_suite_info):
+                rosie_info = {}
+                try:
+                    for line in open(rosie_suite_info, 'r').readlines():
+                        rosie_key, rosie_val = line.strip().split("=", 1)
```

Comment:
Trying this out with some real-world cases I encountered this traceback:

```
Traceback (most recent call last):
  File "/net/home/h06/osanders/cylc-2/lib/cherrypy/_cprequest.py", line 679, in respond
    response.body = self.handler()
  File "/net/home/h06/osanders/cylc-2/lib/cherrypy/lib/encoding.py", line 230, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/net/home/h06/osanders/cylc-2/lib/cherrypy/_cpdispatch.py", line 60, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/net/home/h06/osanders/cylc-2/lib/cylc/review.py", line 471, in suites
    rosie_key, rosie_val = line.strip().split("=", 1)
ValueError: need more than 1 value to unpack
```

To replicate try viewing a suite list for a user who runs Rosie suites.

Reply:
It worked for my own Rosie suites, but good spot; looking at the code this would only occur for suites with a ``rose-suite.info`` whereby, other than invalid lines, there are non-setting lines. There aren't any sections for ``rose-suite.info``, so for a valid file really just any comment lines, which I didn't anticipate, though if they contain an ``=`` they would erroneously be parsed. I'll adapt the code to catch these cases.